app/controllers/lessons_controller.rb

```diff
@@ -87,7 +90,10 @@ def update
         format.html { redirect_to @lesson, notice: 'Lesson was successfully updated.' }
         format.json { head :no_content }
       else
-        format.html { render action: "edit" }
+        format.html do
+          @venues = current_school.venues
+          render action: "edit"
+        end
```

Comment:
@venues is used once in the form for lessons, so it should be possible to add a line in the form partial to replace all the `@venues =` used in the controller. Testing it...